### PR TITLE
Vehicle spawn and fill with template crew functions .

### PR DESCRIPTION
**Pull request by CalebSerafin**
## What type of PR is this.
* Bug
* Change
3. [x] Enhancement

### A3A_fnc_spawnVehicle 
* _A3-Antistasi\functions\CREATE\fn_spawnVehicle.sqf_
* Vehicle spawning takes position and angle in multiple formats.
* Vehicle spawning has features if BIS_fnc_spawnVehicle
* Vehicle spawning can be further customised that BIS_fnc_spawnVehicle.

### A3A_fnc_spawnVehicleCrew 
* _A3-Antistasi\functions\CREATE\fn_spawnVehicleCrew.sqf_
* Vehicle Crew spawning randomly selects crew from templates if possible.
* Vehicle Crew spawning does not create unnecessary groups
* Vehicle Crew spawning fills up all non-cargo seats.

### A3A_fnc_setPos 
* _A3-Antistasi\functions\Utility\fn_setPos.sqf_
* Set position of object using different coordinate systems.

## Following steps after merge
* Get someone who knows the implementation of the new template system to add that compatibility into the marked locations in source code.
* Modify safeVehicleSpawn to work with all custom angles and position systems, so we can use that instead.
* Integrate were possible into Wurzel Hardcore Support

## Please specify which Issue this PR Resolves.
closes https://github.com/official-antistasi-community/A3-Antistasi/issues/1368

## Please verify the following and ensure all checks are completed.

2. [x] Have you loaded the mission in LAN host?
3. [x] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
* Yes

### How can the changes be tested?
Spawn rebel crewed LSV in proximity to the player.
```sqf
["B_T_LSV_01_armed_F",getPos player, 0, resistance, 20] call A3A_fnc_spawnVehicle;
```
Spawn rebel crewed LSV on top roof above player.
```sqf
private _myPos = getPosWorld player;
_myPos = [_myPos#0,_myPos#1,0,"AGLS"];  // Spawn 0m above highest roof above the player.
["B_T_LSV_01_armed_F",_myPos, 0, resistance, 20] call A3A_fnc_spawnVehicle;
```
Spawn rebel crewed ghost-hawk part of your squad at a low turn above player.
```sqf
private _myPosW = getPosWorld player;
private _vectorDirAndUp = [[0.676148,-0.736273,-0.0269321],[-0.571476,-0.547179,0.611565]];
private _vehicle = ["B_Heli_Transport_01_F",[_myPosW,"WORLD"], _vectorDirAndUp, group player, nil, [10,40]] call A3A_fnc_spawnVehicle;
_vehicle enableSimulation false; // For admiring the artwork

// Look and execute at ghost-hawk to release.
cursorObject enableSimulation true;
```

### CODE REVIEW

**Comment by CalebSerafin** [comment]
Okay, I'll delete the last squash commit out of history, thinks I added stuff.

File: A3-Antistasi/functions/CREATE/fn_spawnVehiclePrecise.sqf [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    _velocity = _aircraftPhysics#1;
+};
+private _createVehicleSpecial = ["CAN_COLLIDE","FLY"] select (_addAircraftPhysics);
+
+private _vehicle = objNull;
+if (isNil { // Make sure vehicle is spawned and placed within the same physics step.
+    _vehicle = createVehicle [_className, _position select [0,2], [], 0, _createVehicleSpecial];
+    if (isNull _vehicle) then {
+        [1, "InvalidObjectClassName | """+_className+""" does not exist or failed creation.", _filename] remoteExecCall ["A3A_fnc_log",2,false];
+        _vehicle = createVehicle [_className, _position select [0,2], [], 0, _createVehicleSpecial];     // Retry with a vehicle so that a convoy/mission which doesn't check won't error out.
```
Comment (HakonRydland) [inline A3-Antistasi/functions/CREATE/fn_spawnVehiclePrecise.sqf:75]: when would retrying creation with an invalid classname ever not fail?

**Review by HakonRydland — CHANGES_REQUESTED** [review]
mostly good, one change needed and one question

File: A3-Antistasi/functions/CREATE/fn_spawnVehiclePrecise.sqf (hunk 1 above)
Comment (CalebSerafin) [reply A3-Antistasi/functions/CREATE/fn_spawnVehiclePrecise.sqf:75]: That should've been a constant stock vehicle. Will fix.

File: A3-Antistasi/functions/CREATE/fn_spawnVehiclePrecise.sqf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+_vehicle enableSimulation false; // For admiring the artwork
+// Release.
+cursorObject enableSimulation true;
+*/
+params [
+    ["_className","",[ "" ]],
+    ["_position",[0,0,0],[ [] ], [2,3,4]],
+    ["_direction",0,[ 0,[] ], [ 3,2 ]],
+    ["_emptyPositionRadius",0, [ 0 ]],
+    ["_aircraftPhysics",[], [ true,[] ]],
```
[the remark this answers is not in the document]
Comment (CalebSerafin) [reply A3-Antistasi/functions/CREATE/fn_spawnVehiclePrecise.sqf:48]: Ah, should swap that and add data validation if an empty array is issued.

**Review by HakonRydland — APPROVED** [review]
👍

**Pull request merged**